File: Interpreter/DegreesFormatter.java


import android.content.Context

class DegreesFormatter  extends ComplexFormatter {
	DegreesFormatter(Context _context){
		super(_context)
	}
	
	String toString(Complex complex){
		if(complex.i == 0){
			return  toString(math.toDegrees(complex.r), 12) + context.getText(R_string.character_degree)
		}else{
			RadiusAngleFormatter formatter = new RadiusAngleFormatter(context, true)
			return formatter.toString(complex)
		}
	}
}

File: Interpreter/DegreesFormatter.py
from Interpreter.ComplexFormatter import *
from Interpreter.R_string import *
from Interpreter.RadiusAngleFormatter import *

class DegreesFormatter(ComplexFormatter):
	def __init__(self, _context):
		ComplexFormatter.__init__(self, _context)

	def toString(self, complex):
		if complex.i:
			return self.toStringV(math.degrees(complex.r), 12) + R_string.character_degree
		else:
			formatter = RadiusAngleFormatter(self.view, True)
			return formatter.toString(complex)
